src/common/hobject.h

```diff
@@ -300,8 +300,13 @@ WRITE_CLASS_ENCODER(hobject_t)
 namespace std {
   template<> struct hash<hobject_t> {
     size_t operator()(const hobject_t &r) const {
-      static rjhash<uint64_t> I;
-      return r.get_hash() ^ I(r.snap);
+      static std::hash<object_t> H;
+      static rjhash<uint64_t> RJ;
+      size_t hash;
+      hash = H(r.oid);
+      hash = RJ(hash ^ r.get_hash());
```

Comment:
this looks like overkill... the hobject's hash value is the rjenkins hash already, so adding the weak linux hash on top of that shouldn't add any value.

I think this should be sufficient:
<pre>
    size_t operator()(const hobject_t &r) const {
      static rjhash<uint64_t> RJ;
      return RJ(r.get_hash() ^ r.snap);
    }
</pre>

Reply:
Thanks for the suggestion. I've changed the code and will report back tomorrow if it works.

(the problem unfortunately takes its time to show up)

Reply:
@liewegas I used both H(r.oid) and r.get_hash() because I observed that we tend to set hash from network data. If we trust peers, this is obviously overkill.